PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeploymentState.java

```diff
@@ -173,6 +196,52 @@ protected void updateTelemetryMap(@NotNull Map<String, String> telemetryMap) {
         telemetryMap.putAll(springCloudDeployConfiguration.getModel().getTelemetryProperties());
     }
 
+    private void validateSpringBootApp(String finalJar) throws AzureExecutionException, IOException {
+        final JarFile jarFile = new JarFile(finalJar);
+        final Attributes maniFestAttributes = jarFile.getManifest().getMainAttributes();
+        final String mainClass = maniFestAttributes.getValue(MAIN_CLASS);
+        final String library = maniFestAttributes.getValue(SPRING_BOOT_LIB);
+        if (StringUtils.isEmpty(mainClass)) {
+            throw new AzureExecutionException(MAIN_CLASS_NOT_FOUND);
+        }
+        if (StringUtils.isEmpty(library)) {
+            throw new AzureExecutionException(FAILED_TO_LOAD_DEPENDENCIES);
+        }
+        final Map<String, String> dependencies = getSpringAppDependencies(jarFile.entries(), library);
+        if (!dependencies.containsKey(SPRING_BOOT_AUTOCONFIGURE)) {
+            throw new AzureExecutionException(NOT_SPRING_BOOT_PROJECT);
+        }
+        final String springVersion = dependencies.get(SPRING_BOOT_AUTOCONFIGURE);
+        for (String artifact : SPRING_ARTIFACTS) {
+            if (!SpringCloudDependencyManager.isCompatibleVersion(dependencies.get(artifact),
+                                                                  springVersion)) {
+                PluginUtil.showInfoNotificationProject(project, DEPENDENCIES_IS_NOT_UPDATED,
+                                                       String.format(DEPENDENCY_WARNING, artifact));
+                return;
+            }
+        }
+    }
+
+    private Map<String, String> getSpringAppDependencies(Enumeration<JarEntry> jarEntryEnumeration,
+                                                         String libraryPath) {
+        final List<JarEntry> jarEntries = Collections.list(jarEntryEnumeration);
+        return jarEntries.stream()
+                         .filter(jarEntry -> StringUtils.startsWith(jarEntry.getName(), libraryPath)
+                                 && StringUtils.endsWith(jarEntry.getName(), JAR))
+                         .map(jarEntry -> {
+                             String fileName = StringUtils.strip(jarEntry.getName(), libraryPath);
+                             fileName = StringUtils.stripEnd(fileName, JAR);
+                             final int i = StringUtils.lastIndexOf(fileName, "-");
```

Comment:
StringUtils.split("ab:cd:ef", ":", 2)    = ["ab", "cd:ef"]

Reply:
StringUtils.split could set split group number, but seems unable to split from the last appearence